### ISSUE TITLE
Dynamic bayesian networks (DBNs)

### ISSUE THREAD

**Issue by aditya1709** [biopython/biopython#1836]
Has the Mocapy++ been integrated into biopython. I can't seem to find any sample code on DBNs in biopython. Any help is appreciated. [I am specifically looking at parameter learning in DBNs]

**Comment by peterjc** [comment]
No, it has now. Are you talking about https://sourceforge.net/projects/mocapy/ which is a C++ project? What form would such an integration take?

**Comment by aditya1709** [comment]
There was a GSoC 2011 project that was focused on integrating Mocapy++ into biopython. Wanted to confirm if it was done.

**Comment by peterjc** [comment]
You can read about that on https://biopython.org/wiki/GSOC2011_Mocapy (although it looks like some of the Mediawiki markup broke in the move to Markdown on GitHub pages). There is some code on https://github.com/mchelem/biopython/commits/master (yes, the fork's master branch), but I don't recall anything being merged into the official Biopython repository.

Hopefully @mchelem or @etal can clarify where things stand?

**Comment by mchelem** [comment]
A couple of years ago @etal contacted Thomas Hamelryck (my GSoC mentor) and I to discuss integrating Mocapy's fork into the official Biopython repository. Back then, Thomas wanted to include some new work a masters student under his supervision was finishing. After that, we didn't keep in touch. 

I just checked the Mocapy project and noticed the GSoC branch has been integrated into the main source tree: https://sourceforge.net/p/mocapy/code/HEAD/tree/Mocapy/python/

@aditya1709 My Biopython fork is quite old, but if you are willing to try it, the following is a good example on how to train a model: https://github.com/mchelem/biopython/blob/master/Bio/PDB/TorusDBN/TorusDBNTrainer.py
